docs2/site/docs/migrations/migration4.md

```diff
@@ -59,40 +103,93 @@
 * `ISchema.FindDirective`, `ISchema.RegisterDirective`, `ISchema.RegisterDirectives` methods were moved into `SchemaDirectives` class
 * `ISchema.FindType` method was moved into `SchemaTypes[string typeName]` indexer
 * Some of the `ISchemaNodeVisitor` methods have been changes to better support schema traversal
+* `SourceLocation`, `NameNode` and `BasicVisitor` have been changed to a `readonly struct`.
+* `ObjectExtensions.GetInterface` has been removed along with two overloads of `GetPropertyValue`.
+* `void Visit<TState>(System.Action<INode, TState> action, TState state)` method has been added.
+* Various `IEnumerable<T>` properties on schema and graph types have been changed to custom collections: `SchemaDirectives`, `SchemaTypes`, `TypeFields`, `PossibleTypes`, `ResolvedInterfaces`
+* `INode.IsEqualTo` and related methods have been removed.
+* `ApolloTracing.ConvertTime` is now private and `ResolverTrace.Path` does not initialize an empty list when created.
+
+### `ExecutionOptions.EnableMetrics` is disabled by default
+
+To enable metrics, please set the option to `true` before executing the query.
+
+(todo: add sample)
+
+### GraphQL Member Descriptions
+
+> By default, descriptions for fields, types, enums, and so on are not pulled from xml comments unless the corresponding global flag is enabled.
+
+Note that to improve performance, by default GraphQL.NET 4.0 does not pull descriptions from fields from xml comments as it did in 3.x. To re-enable that functionality, see Global Switches above.
+
+(todo: update/consolidate description)
+
+### Changes to `IResolveFieldContext.Arguments`
+
+`IResolveFieldContext.Arguments` now returns an `IDictionary<string, ArgumentValue>` instead of `IDictionary<string, object>` so that it
+can be determined if the value returned is a default value or if it is a specified literal or variable.
+
+`IResolveFieldContext.HasArgument` now returns `false` when `GetArgument` returns a field default value. Note that if a variable is specified,
+and the variable resolves to its default value, then `HasArgument` returns `true` (since the field argument is successfully resolving to a variable
+specified by the query).
+
+### Metadata is Not Thread Safe
+
+> `IProvideMetadata.Metadata` is now `Dictionary` instead of `ConcurrentDictionary`, and is not thread safe anymore
+
+(todo: update description, add description of how to lock on type, not dictionary)
+
+### Other Breaking Changes
+
+* GraphQL.NET now uses GraphQL-Parser v7 with new memory model
+* When used, Apollo tracing will now convert the starting timestamp to UTC so that `StartTime` and `EndTime` are properly serialized as UTC values.
+* `Connection<TNode, TEdge>.TotalCount` has been changed from an `int` to an `int?`. This allows for returning `null` indicating that the total count is unknown.
+* `InputObjectGraphType.ParseDictionary` has been added so that customized deserialization behavior can be specified for input objects.
```

Comment:
```suggestion
* `InputObjectGraphType.ParseDictionary` has been added so that customized deserialization behavior can be specified for input objects (aka input resolvers).
```